doc/SONiC_long_reset_button_press_hld.md

```diff
@@ -0,0 +1,282 @@
+# Long reset button press HLD
+
+## Table of contents
+
+-   [Revision](#)
+-   [Scope](#scope)
+-   [Definitions/Abbreviations](#definitionsabbreviations)
+-   [Requirements](#requirements-in-this-feature-we-want-to-be-able-to)
+-   [Architecture design](#architecture-design)
+-   [High level design](#high-level-design)
+-   [Performance](#performance)
+-   [Flows](#flows)
+-   [Configuration and management](#configuration-and-management)
+    - [CLI](#cli)
+    - [YANG Model](#yang-model)
+-   [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+-   [Restrictions/Limitations](#restrictionslimitations)
+-   [Testing Requirements/Design](#testing-requirementsdesign)
+    - [Unit Test cases](#unit-test-cases)
+    - [System Test cases](#system-test-cases)
+
+#
+  
+
+# Revision
+
+| **Rev**   | **Date**     |           **Author**            | **Change Description**  |
+|:---------:|:------------:|:-------------------------------:|:------------------------|
+| 1.0       | 04/01/2024   | Azmy Ali (Nvidia SONiC Team)    | Initial public version  |
+
+### 
+
+# Scope
+
+This document describes the design details for supporting long reset button press on a SONiC switch. In this document, we will specify the default behavior
+for SONiC OS on long reset button press and the new service introduced during initializtion flow to handle the functionality applied on long reset button press.
+
+# Definitions/Abbreviations
+
+| **Term**                    | **Description**                                                                      |
+|-----------------------------|--------------------------------------------------------------------------------------|
+| SONiC | Software for Open Networking in the Cloud |
+|   Long reset button press   |   Pressing the physical reset button on the switch not less than 15 seconds          |
+|   short reset button press  |   Pressing the physical reset button on the switch less than 15 seconds              |
+| CLI   | Сommand-line Interface                    |
+| YANG  | Yet Another Next Generation               |
+
+### 
+
+# Requirements Overview
+
+The long reset button press feature implementation shall support the following:
+
+    1. On long reboot press, the OS will restore the local users' configurations, by deleting non-default users and restoring default passwords for default users and expiring  these passwords due to security conecerns.
```

Comment:
Is the use-case mainly to reset the password of the default user? if no, why cant we login to the switch with the default user to delete the non-default users explicitly?

Reply:
yes, I changed the feature name accordingly. the fearture will be reset local users for the default users and in the default implementation this will be triggered by a long reset button